### PR TITLE
Revise Windows build instructions

### PR DESCRIPTION
**Pull request by wrowe**
Commit Message: Revise Windows build instructions

Additional Description:
- No longer limited to building envoy-static
- Explain the c:\c -> c:\ symlink and TMPDIR more clearly
- Explain output_base path limitations
  https://github.com/envoyproxy/envoy/issues/12098
- General wordsmithing

Risk Level: n/a
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Fixes 12098

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @envoyproxy/windows-dev

**Comment by sunjayBhatia** [comment]
@rmiller14

**Comment by sunjayBhatia** [comment]
I'll wait to stamp this as done for @rmiller14 to weigh in with any thoughts since youre getting familiar with the build setup

File: bazel/README.md [hunk 1]
```diff
@@ -112,12 +112,29 @@ for how to update or override dependencies.
...
 
-    [python3](https://www.python.org/downloads/): Specifically, the Windows-native flavor. The POSIX flavor
-    available via MSYS2 will not work, nor will the Windows Store flavor. You need to add a symlink for `python3.exe` pointing to
-    the installed `python.exe` for Bazel rules which follow POSIX conventions. Be sure to add
-    `pip.exe` to the PATH and install the `wheel` package.
+    Follow all items below to configure the Windows build toolchain and prepare the build environment,
+    or just install Docker, Git, MSYS2 to use the Docker image (the same as used in the CI pipeline):
+    ```
+    ./ci/run_envoy_docker_windows.sh './ci/windows_ci_steps.sh'
+    ```
```
Comment (nigriMSFT) [inline bazel/README.md:120]: It looks like this method already has its own section "Building Envoy with the CI Docker image", but is lacking the extra dependency information. Is this supposed to be a recommendation that developers use the docker image?

Comment (sunjayBhatia) [reply bazel/README.md:120]: I was thinking we add a note at the top of the "bazel quick start" section that links to the docker section so people who want to use docker can short circuit all the toolchain setup (and also add the dependencies needed for docker down there, will push a change that reflects that

**Comment by nigriMSFT** [comment]
I added these Windows example setup command steps assuming bazel is invoked via cmd or powershell. This might be stale and/or not recommended already? From what I've heard I think everyone (including CI) invokes bazel from MSYS except me.

**Comment by sunjayBhatia** [comment]
> I added these Windows example setup command steps assuming bazel is invoked via cmd or powershell. This might be stale and/or not recommended already? From what I've heard I think everyone (including CI) invokes bazel from MSYS except me.

We could make it clear in each instance that these are examples of how to set the PATH in `cmd` specifically and assume developers will adapt accordingly, change the examples to bash syntax, or omit the examples entirely?

Meta point around bash/MSYS2, we decided to use it in CI scripts/tooling scripts b/c it would be most familiar for the current set of core maintainers/most contributors, if that decision no longer makes sense we can revisit it

**Comment by rmiller14** [comment]
@sunjayBhatia Sorry for the delay.  I'll submit a PR to add some clearer instructions on top of what was already submitted.  The biggest confusion for me was around paths and in which shell to build, as well as the tips for Windows.  I also ended up having to enable symlink support as specified in the bazel documentation - I would call that out explicitly.

The big gotcha for me was that the ordering of the entries in the path environment do matter, as link.exe can refer to either the one that's part of VS builds tools or the one that's in /path/to/msys/usr/bin.

It seems like it's ok to build in any of cmd, powershell, or msys at the moment.  I ended up building from powershell.  I would call that out explicitly too.

**Pull request merged**